### DISCUSSION TITLE
How to create app.vue file and make it as parent of layouts

### DISCUSSION THREAD

**Discussion by saifobeidat** [nuxt/nuxt#10182] [Questions (Nuxt 2)]
Hello,

I have multiple layouts, and I have some logic that's executed on vue instance creation, I put it in the default.vue layout, but I had to duplicate it in the other layouts..

this is an example of the duplicated logic:

```
created() {
    this.$i18n.setLocale(this.$route.params.lang);
  },
```
how can I handle this and put in in the app level without having to duplicate it? I couldn't know how to add app.vue file which will behave as a parent of  the layouts.. is there a way to handle this thing?

**Comment by benjaminrobinet** [comment]
An easy way is the create a mixin called "LayoutMixin.js" for exemple, and you will add to all your layouts this mixin.

I think that you can also create a plugin that uses a Navigation Guard for this need.
